qdev_wrappers/alazar_controllers/acquisition_parameters.py

```diff
@@ -63,7 +62,7 @@ def check(self):
         Return:
             True (if no errors raised when check_and_update_fn executed)
         """
-        val = self._latest['value']
+        val = self.cache.get()
```

Comment:
in case these parameters are (or happen to be) always such that they `value`s are always equal to their `raw_value`s, then this can be implemented in a backwards compatible manner via `val = self.raw_value`. But if not, then i'd rethink the purpose of this method - as far as i understood it, it basically does the same as `set_raw` but it takes the value to validate from "cache" as opposed to "from outside" - i wonder what the benefit of this functionality is, because thanks to `set_raw` the value in "cache" can't be not-valid, right?